templates/new/batch_main.py.txt

```diff
@@ -1,22 +1,57 @@
 import argparse
-from surround import Assembler
+from surround import Assembler, RunMode, Config
```

Comment:
I still think there are too many details here that don't need to be exposed. I've got a small example below showing a simplified version, the `Surround` class would be part of the `surround` project. There would probably need to be a `--help` flag that printed out the options for runners, modes and assemblies. I also think that `Web` should be a runner rather than a specific mode as we may want to trigger a batch predict or a train from a web application. 

```
import argparse
from surround import Assembler, RunMode, Config
from stages import Main, InputValidator
from batch_runner import FileSystemRunner

assemblies = [
    Assembler("baseline")
        .set_validator(InputValidator())
        .set_estimator(Main())
]


class Surround:

    def __init__(self, assemblies):
        self.assemblies = assemblies


    def run(self, runner, assembler_key, mode):
        assembler = self.__get_assembler(assembler_key)
        if assembler:
            runner(assembler).run(mode)
        else:
            print("Unable to find the assembler '%s'" % assembler_key)


    def __get_assembler(self, key):
        assembler = None

       # Try and get the assembler via index
        try:
            index = int(key)

            if 0 <= index < len(self.assemblies):
                assembler = self.assemblies[index]
        except Exception:
            pass

       # Try and get the assember via key (name)
        if not assembler:
            assembler = next((x for x in self.assemblies if x.assembler_name.lower() == key.lower()), None)

        return assembler

def main():
    config = Config(auto_load=True)
    default_assembler = config.get_path('assembler.default')

    parser = argparse.ArgumentParser(prog='Pizza', description="Surround mode(s) available to run this module")
    parser.add_argument('--runner', help="Runner for the application", default=0)
    parser.add_argument('--assembler', help="Assembler to run (index or name)", default=default_assembler if default_assembler is not None else "0")
    parser.add_argument('--mode', help="Mode to run (train, batch)", default="batch")

    args = parser.parse_args()
    s = Surround([FileSystemRunner], assemblies)
    s.run(args.runner, args.assembler, args.mode)

if __name__ == "__main__":
    main()
```